exporter/otlpexporter/connection.go

```diff
@@ -0,0 +1,158 @@
+// Copyright  OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package otlpexporter
+
+import (
+	"context"
+	"math/rand"
+	"sync/atomic"
+	"time"
+	"unsafe"
+
+	otlpmetriccol "github.com/open-telemetry/opentelemetry-proto/gen/go/collector/metrics/v1"
+	otlptracecol "github.com/open-telemetry/opentelemetry-proto/gen/go/collector/trace/v1"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+)
+
+// Connection implementation is borrowed and adapted from OpenTelemetry Go SDK implementation:
+// https://github.com/open-telemetry/opentelemetry-go/blob/e0406dd3eb7aa826aba19287e8a14ca016bf8578/exporters/otlp/connection.go#L15
+
```

Comment:
I was not aware that making a connection is so much code, can you explain what exactly do we try to achieve?

Reply:
This attempts to connect and reconnect on errors. Code copied from OpenTelemetry Go SDK, not written by me.

Reply:
I thought that grpc handles reconnects. Maybe I am wrong.

Reply:
This reminds me of the OCAgent code that wanted to keep a connection because it used bi-di stream, likely we can simplify this down the line.